### DISCUSSION TITLE
OAuth setup with Synology SSO

### DISCUSSION THREAD

**Discussion by JoTec2002** [mealie-recipes/mealie#4667] [OAuth Provider Example]
### Configuration Example

**Prerequisits:**
On your Synology NAS System you need to have the package "SSO Server" installed and complete the general configuration according to the guide from Synology. Furtermore the OIDC Service needs to be enabled. [Guide to Synology SSO Server](https://kb.synology.com/de-de/DSM/help/SSOServer/sso_server_general_setting?version)

**Settings on Synology NAS:**
1. Create new Application in your SSO Server
![Step 1](https://github.com/user-attachments/assets/098a0768-55d5-4ab3-a7e4-3117e019359d)
2. Select OIDC as SSO-Protocol
![Step 2](https://github.com/user-attachments/assets/d97b53ae-06c5-46e3-b1f4-fa933c00e4bc)
3. Enter your application name (can be freely chosen). Enter the correct Redirect URL. Needs to be https://<YOUR_MEALIE_URL>/login. Click finish
![Step 3](https://github.com/user-attachments/assets/b8cd894a-63dd-43ba-a136-03501b01e6cc)
4. Click on edit for you newly created Application.
![Step 4](https://github.com/user-attachments/assets/d2fffbee-255b-4c4a-889e-df2be46f09d4)
5. Make note of your ID & Secret
![Step 5](https://github.com/user-attachments/assets/856e92f6-b523-4f21-862b-1db6318417c4)

**Settings on in your Mealie Config:**
1. Go to your mealie docker-compose.yml file and add the following to environment:
```
#OIDC
OIDC_AUTH_ENABLED: "True"
OIDC_CONFIGURATION_URL: "<URL FROM your Synology SSO Server>" # Can be found on the Synology under SSO Server>Service>OIDC>Well-known URL
OIDC_CLIENT_ID: "<Auto generated ID from above>"
OIDC_CLIENT_SECRET: "Auto generated Secret from above "
OIDC_PROVIDER_NAME: "Synology NAS" #Can be freely chosen 
OIDC_NAME_CLAIM: "username" #needs to be set to username
```
2. Recreate your container and voilà, you're done

**Comment by b42n1** [comment]
I attempted to use the wildcard with /login* as [described in the documentation](https://docs.mealie.io/documentation/getting-started/authentication/oidc/#provider-setup), but it consistently resulted in an error.

**Comment by b42n1** [comment]
If you have existing accounts with the same email address, logging in through Synology will access those accounts. To switch the authentication method to only OIDC, you need to update the authentication method from Mealie to OIDC in the account settings under `/admin/manage/users`. After making this change, you will only be able to log into these accounts using OIDC. However, if you retain the default Mealie authentication method, both the original login credentials and OIDC will remain available.

![image](https://github.com/user-attachments/assets/ee56a291-caa9-47ad-9629-0608c858e595)

**Comment by gregorne** [comment]
I had the issue of mealie not being able to resolve the Synology SSO domain since it was running in a container on the same host.

Adding:

extra_hosts:
      - "your-domain.synology.me:192.168.X.X"
 
to the stack (docker compose yaml) resolved the issue

Note:
 - replace your-domain.synology.me with your actual DDNS donain used by Synology SSO Server.
 - replace 192.168.X.X with the local ip of your NAS

Also use proper indentation i cant get the formatting right in a comment
